src/sentry/migrations/0269_delete_legacy_alert_rule_action_records.py

```diff
@@ -0,0 +1,97 @@
+# Generated by Django 2.2.24 on 2022-01-27 00:00
+
+import pickle
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+from sentry.utils.strings import decompress
+
+
+def delete_metric_alert_trigger_actions_with_alert_rule_actions(apps, schema_editor):
+    """
+    Delete the rows in the AlertRuleTriggerAction table that use an Alert Rule Action SentryApp.
+    """
+    AlertRuleTriggerAction = apps.get_model("sentry", "AlertRuleTriggerAction")
+    SentryAppComponent = apps.get_model("sentry", "SentryAppComponent")
+
+    for trigger_action in AlertRuleTriggerAction.objects.filter(
```

Comment:
I think there are two ways to do this query more efficiently:
```py
query = AlertRuleTriggerAction.objects.filter(
    sentry_app__type="alert-rule-action",
    sentry_app_config__isnull=False,
)

for trigger_action in query:
    ...
```
```py
query = AlertRuleTriggerAction.objects.all().select_related("sentry_app")
for alert_rule_trigger_action in RangeQuerySetWrapperWithProgressBar(query):
    if (
        alert_rule_trigger_action.sentry_app.type == "alert-rule-action" 
        and alert_rule_trigger_action.sentry_app_config is not None
    ):

```
You'd have to try them out though I haven't tested this.

Reply:
@mgaeta the `sentry_app_id` column in `sentry_alertruletriggeraction` is a FK to `sentry_sentryapp` not `sentry_sentryappcomponent`. `type` is stored in `sentry_sentryappcomponent`